### DISCUSSION TITLE
Add a third level to the navigation menu

### DISCUSSION THREAD

**Discussion by vv-ac** [filamentphp/filament#2211] [Ideas]
Hi,

as known from Laravel Nova it would be great to have a third layer to structure the navigation menu.
See for reference: https://nova.laravel.com/docs/4.0/customization/menus.html#building-menus
Nova provides:
- Section
- Group
- Item

This would add more possibility to depcit the application structure to the user.

**Comment by raftalks** [comment]
This is nice idea to add. However I am more concerned that the sidebar will be of with too many nav items. How about we have section as like isolated apps or modules for filament? The sidebar can have a dropdown to navigate into different sections, and on each section the resources displayed will be those that belong to that section. For example, Settings section can have multiple groups like General Settings, Security, etc and under them their relevant resources.

**Reply by vv-ac** [reply]
I am not too much of a fan when it comes to dropdowns in the sidebar except for we talk about collapsible sections, then it's a yes from me.

That is also what NOVA is doing.
![grafik](https://user-images.githubusercontent.com/67121752/167089806-e0465477-52af-47e9-a38b-eeeff86a46a5.png)

I used it before and actually felt that the sidebar gets more clarity from that as you can clearly distinguish between some "admin backend" and some "frontend" sections in Filament, especially for users with extensive access.

**Reply by pxlrbt** [reply]
Never liked how everything in Nova was thrown under "Resources".

**Reply by vv-ac** [reply]
Same, not meaning that things should be grouped by standard, just a third (collapsible) level to be added.

**Comment by uadoronin** [comment]
Please, make third-level navigation, it is a really necessary thing.

**Reply by pxlrbt** [reply]
Could you provide an example how you would use a 3rd level?

**Reply by vv-ac** [reply]
To me it is about seperating major page areas from each other.
E. g. have a section "Administration" which simply deals with configuration/ customizing and a more content focused area.

This comes in handy in particular when you have Filament to be the acutal application rather than "just" an administration backend: https://github.com/filamentphp/filament/discussions/2198

**Reply by danharrin** [reply]
I'm afraid that if we add a 3rd level, it will take 1 week before there is a new discussion "Add a fourth level to the navigation menu" 😆

**Reply by vv-ac** [reply]
Couldn't we have a megamenu? :) Just kidding.
I think 3 is a good match, didn't see a discussion over at nova to have a fourth level ;-)

**Reply by uadoronin** [reply]
I experimented and decided. You're right. The groups is quite enough for menu navigation. You'd better work on the other features, thank you.

**Reply by MarJose123** [reply]
I think "Add a fourth level to the navigation menu" don't make any sense anymore as it will be achieved by 3rd level and I think there are no 4th level in real scenario.

**Comment by ManojKiranA** [comment]
This is now possible see #4650

**Reply by MarJose123** [reply]
this https://github.com/filamentphp/filament/pull/4650 will only achieve for doing advance, however if you are using resource menu you can't achieve this.

**Reply by ManojKiranA** [reply]
You mean menu generated by resource class.

**Reply by ManojKiranA** [reply]
You can overide the entire navigation builder and configure how you want.
